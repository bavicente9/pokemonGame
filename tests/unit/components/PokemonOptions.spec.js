
import {shallowMount} from '@vue/test-utils'

import PokemonOptions from '@/components/PokemonOptions'

import {pokemons} from '../mocks/pokemons.mock'

describe('PokemonOptions component', () => {
  
    let wrapper 
    
    beforeEach(() => {
        
         wrapper = shallowMount(PokemonOptions,{
            props:{
                pokemons
            }
         })
    })

    test('Snapshot match',() => {

        expect(wrapper.html()).toMatchSnapshot()
    })

   

    test('must emit "selection" with its params when is clicked',() => {
      const [li1,li2,li3,li4] = wrapper.findAll('li')
    
      li1.trigger('click')
      li2.trigger('click')
      li3.trigger('click')
      li4.trigger('click')
      
      expect(wrapper.emitted('selection').length).toBe(4)
      
      expect(wrapper.emitted('selection')[0]).toEqual([1])
      expect(wrapper.emitted('selection')[1]).toEqual([2])
      expect(wrapper.emitted('selection')[2]).toEqual([3])
      expect(wrapper.emitted('selection')[3]).toEqual([4])
    })
    
})