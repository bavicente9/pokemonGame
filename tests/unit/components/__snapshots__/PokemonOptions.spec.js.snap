// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions component Snapshoot match 1`] = `
<h2>Pokemon options</h2>
<div class="options-container">
  <ul>
    <li>bulbasaur</li>
    <li>ivysaur</li>
    <li>venusaur</li>
    <li>charmander</li>
  </ul>
</div>
`;
